docs/blog/_posts/2018-03-05-seventh-dotty-milestone-release.md

```diff
@@ -162,7 +162,7 @@ object Interpreter {
 }
 ```
 
-Earlier Dotty releases had issues typechecking such interpreters.
+Earlier Scalac and Dotty releases had issues typechecking such interpreters.
```

Comment:
This makes it sound like recent releases of scalac have fixed this, but they haven't. Also I would use the term "Scala 2" and not "Scalac".

Reply:
The most convincing thing to do would be to give an example that doesn't typecheck in Dotty but does in Scala 2.

Reply:
This program works in Scalac, while trickier ones trigger bugs in Scalac or Dotty. I propose to write "Scala 2 and Dotty have issues [...]", since we fixed some.

```scala
case class MapS[S, T](xs: Exp[List[S]], f: Exp[S => T]) extends Exp[List[T]]
case class Map[S, T, That](xs: Exp[List[S]], f: Exp[S => T])(implicit val cbf: CanBuildFrom[List[S], T, That]) extends Exp[That]

//...

    case m @ Map(xs, f)    =>
      //val f1: Exp[Any => Any] = f //accepted by Scalac, rejected by Dotty!
      eval(xs).map(eval(f))(m.cbf)

    case MapS(xs, f) =>
      eval(xs) map eval(f) //rejected by Dotty, not 100% sure why. Passing the 

   //alternatively:
    case m: MapS[s, t] =>
      //(eval(m.xs) map eval(m.f))(List.canBuildFrom) //fails in Dotty
      (eval(m.xs) map eval(m.f))(List.canBuildFrom[t]) //succeeds in Dotty

```

Failures in https://github.com/lampepfl/dotty/pull/4069 might be relevant.